internal/tasks/templateversion.go

```diff
@@ -65,15 +61,113 @@ type TemplateVersionLogic struct {
 	renderedConfig   *config_latest_types.Config
 }
 
+func NewTemplateVersionLogic(taskManager TaskManager, log logrus.FieldLogger, store store.Store, resourceRef ResourceReference) TemplateVersionLogic {
+	return TemplateVersionLogic{taskManager: taskManager, log: log, store: store, resourceRef: resourceRef}
+}
+
 func (t *TemplateVersionLogic) CreateNewTemplateVersion(ctx context.Context) error {
+	fleet, err := t.store.Fleet().Get(ctx, t.resourceRef.OrgID, t.resourceRef.Name)
+	if err != nil {
+		return fmt.Errorf("fetching fleet: %w", err)
+	}
+	t.fleet = fleet
+
+	err = t.validateFleetTemplate(ctx, fleet)
+	if err != nil {
+		return fmt.Errorf("validating fleet: %w", err)
+	}
+
 	templateVersion := api.TemplateVersion{
 		Metadata: api.ObjectMeta{Name: util.TimeStampStringPtr()},
 		Spec:     api.TemplateVersionSpec{Fleet: t.resourceRef.Name},
 	}
-	_, err := t.store.TemplateVersion().Create(ctx, t.resourceRef.OrgID, &templateVersion, t.taskManager.TemplateVersionCreatedCallback)
+
+	_, err = t.store.TemplateVersion().Create(ctx, t.resourceRef.OrgID, &templateVersion, t.taskManager.TemplateVersionCreatedCallback)
 	return err
 }
 
+func (t *TemplateVersionLogic) validateFleetTemplate(ctx context.Context, fleet *api.Fleet) error {
+	if fleet.Spec.Template.Spec.Config == nil {
+		return nil
+	}
+
+	invalidConfigs := []string{}
+	for i := range *t.fleet.Spec.Template.Spec.Config {
+		configItem := (*t.fleet.Spec.Template.Spec.Config)[i]
+		name, err := t.validateConfigItem(ctx, &configItem)
+		t.log.Debugf("Validated config %s from fleet %s/%s: %v", name, t.resourceRef.OrgID, t.resourceRef.Name, err)
+		if err != nil {
+			invalidConfigs = append(invalidConfigs, name)
+		}
+	}
+
+	condition := api.Condition{Type: api.FleetValid}
+	var retErr error
+	if len(invalidConfigs) == 0 {
+		condition.Status = api.ConditionStatusTrue
+		condition.Reason = util.StrToPtr("Valid")
+		retErr = nil
+	} else {
+		condition.Status = api.ConditionStatusFalse
+		condition.Reason = util.StrToPtr("Invalid")
```

Comment:
Shouldn't this be just Valid true/false ?

Valid + true
Invalid + false

looks inverting, I am looking at k8s api and will ping back here.

I wonder if it makes more sense to keep:

Valid true
Valid false

or 

Valid true
Invalid true 
?

Reply:
Ignore this comment, I was looking wrong at Reason/Type

your logic here is good and correct.